Refactoring

jonnald0430 opened the pull request:
packages and dependencies

dev-alto committed:
⭐ initial
🛠️ removes unused components

dev-alto committed:
🛠️ local cleanup of current components

dev-alto committed:
🪁 added robots.txt
🛠️ cleanup of @/public/assets/
🛠️ swapped some `Techstack` technology icons to `Image`s instead of react-icons for non-flat colors

dev-alto committed:
🪁 added hackathon album to gallery and disposition

dev-alto committed:
🛠️ major project file structure overhaul for Routes, components, and shared components.
🛠️ removed legacy `/p5js` route redirect

dev-alto committed:
🛠️ introduction of improvised podium section and new `Gallery` entries for `/hackathon`
🛠️ single import via _modules system for components

dev-alto committed:
🛠️ redesigned podium section

dev-alto committed:
🛠️ minor cleanup of legacy client logging

dev-alto committed:
fixed instagram link on Overview

dev-alto committed:
🛠️ centered `Navbar` buttons, added faint backgrounds
🛠️ fixed YouTube link
🛠️ expanded socials to `Footer`

dev-alto committed:
🛠️ Hotfix for build error caused by hackathon/[...slug] subpage

dev-alto committed:
🔥🛠️ Fix for node build on Vercel :(
Updated NextJS -> 14.2.5

dev-alto committed:
🛠️ Fixes to image assets including `Home` ~gallery, mascot.gif, diagonal_slowglide_overlay.gif, mediating FCP (First Contentful Paint) load times
🛠️ Fixes to incorrect asset name referencing(jpg->JPG) as seen after building on Vercel, fixing broken media loading on Production
🛠️ Tweaks to wording and visual FX around text to increase readability

dev-alto committed:
🛠️ fixed 404 asset naming

dev-alto committed:
💡 Added max-width to Officer `MajorLeaderCard` card section container to ensure consistent row layout on unconventional screen widths and browser zoom scales
💡 Added new `/promotion` subpage as a temporary promotional inlet landin page
🛠️ Changed div of `Navbar` & `Footer` to respective dedicated html elements for clearer

dev-alto committed:
🛠️ Minor polishings, update to more accurate meeting time

dev-alto committed:
🛠️ Multitude of polishings most `/home` components

dev-alto committed:
🛠️ Refactored `Leadership` section to archive and support the display of past generations of student officers and leadership roles

dev-alto committed:
⚠️ Hotfix for `Leadership` component

jonnald0430 committed:
yarn.lock, package.lock, next-env.d.ts installed

dev-alto commented:
#### 📝 Changed routes:


| [**/hackathon/\[...slug\]**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/hackathon/[...slug]) | [**/hackathon**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/hackathon) |
|:-:|:-:|
|<a href="https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/hackathon/[...slug]"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/9-EY2OsBWzkwDbQiiEahL_dGtxWaBq5_BvUVCnEgN-W8wkg8dA2NDKlnfjbxjieCQgwXD0gxfvdRfyYnG7ZAiQ.png&w=600" alt="Screenshot of /hackathon/[...slug]" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/mu87miqaHVwhj7EopEbJPKeptKZQTzN4GmfcSb7SFgqFwgYLrWf6l75XzVuW74vT2K-omEx8d9ZqkWVkVMe5bg.png">(view full size)</a> |<a href="https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/hackathon"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/nG2c4Wyw8t6JbvEor3qlSU3_sPt57hL4T3PxzDkqWXVjZ75b4-fAsCPK6I4ARlnMBNplqKqxWxpwHNgnVtiGig.png&w=600" alt="Screenshot of /hackathon" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/fFImGXXo7teL3LXe7OvAv7rL0Q4x7Acfy-I9fIPm53BolvTb4o5qNoVe1LGQK6TpMfJWouFDiZ8sFZkx-LH9qw.png">(view full size)</a> |



| [**/home**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/home) | [**/**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/) |
|:-:|:-:|
|<a href="https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/home"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/i6aNTaFuFcpGzLAfrnqVKc83wm8KW5P2hHv-D-9gHdZUw60wZtXt2MbZoKDwM9ZLEIE6SDz3GZF9q7y9msxq_A.png&w=600" alt="Screenshot of /home" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/3xoGwZGulN5hmi5iiBhvczN5CpsetNYPL0F05lCTPmQHXbLRGnSavtUHbPhxXd4TXCR9188ur4MKiRkC8YdpvA.png">(view full size)</a> |<a href="https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/gNxM9b8tGJg3D9eeoNybkRjET5CaaenhpO3NXL53tV3dtwNFEaUlTOly_zIWOHER5izEWAdeMJkWs8VfEAcfqA.png&w=600" alt="Screenshot of /" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/YO3tb4g8OXHnQDdTTdwvA7MYsFoQfqbNmXDsG0pZqdXIeV-4raMh0g-iqyB8BEQlPOI-5GEcOF-Z1U4ulsuVUA.png">(view full size)</a> |



| [**/projects**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/projects) | [**/promotion**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/promotion) |
|:-:|:-:|
|<a href="https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/projects"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/wfgKthRQeEoxzUIyzfr2QKkDobXCTcE89VB_UFlbHo8qVXrU8ZOxBDVM6HnTaUtzWTIZm0VXHexilU2M7UHyDA.png&w=600" alt="Screenshot of /projects" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/6pqD5lPEBfLnnPLXMxTGhJO41xlvwgmJRLhYIiFDVkF6tEVrYvp0ofLQEz3x9gu3zrye5M2QwPvAL01c8cC0cw.png">(view full size)</a> |<a href="https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/promotion"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/Q9Wn9pmzvMX1Gk603mUX-sxlYu_uGi7zS8QUubwjCdm_QHZ8n4SLCCfQtR2xAPy2ssKkoVSEtRTkKM-WfBhpHg.png&w=600" alt="Screenshot of /promotion" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Ubh1mwaZRaNizE320RlXD8vWVnVLzmLYK0ogH7oYqtDxUSYYO9Zt9Je3tK3ixzq95cADYvEkmCL5UDBCBzTAig.png">(view full size)</a> |



And 2 other routes:
- [**/recruitment**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/recruitment)
- [**/resources**](https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app/resources)



Commit 2edd2968a77f0321264a1614fa8ec6d065763b98 (https://the-skycsc-website-jl8b1voyl-skycsc.vercel.app).

dev-alto committed:
⚡Overhaul of many components and sections; improving visual consistency and performance
🌴 Info updating of Leadership section
+ a lot more lost to time (lack of verbosity for this specific commit, sorry QC)

Co-authored-by: CptnMaj <CptnMaj@users.noreply.github.com>

dev-alto committed:
🌴 Overhauled /recruitment page
🔧 Fixed broken event link for meeting notification subscriptions
🔧 Fixed inconsistent scaling for Leadership cards on tall vertical  screens
🔧 Corrected Leadership card information

dev-alto committed:
🔧 Fixed &apos; ESLint build errors
🔧 Fixed array dependency lists for useEffect() scopes of Countdown and /slug components

dev-alto committed:
🌴 Fixed missing submissions on /hackathon
🔧 Attempted fix for broken links to gallery assets

CptnMaj committed:
Added discord and mail to officer cards and removed linktree of club and replaced it with email card instead.

dev-alto commented:
This pull request has gone stale.
